UpskillWikipediaUITests/PageObjects/SearchPage.swift

```diff
@@ -0,0 +1,46 @@
+import Foundation
+import XCTest
+
+class SearchPage: BaseAppPage {
+    
+    // MARK: - Elements
+    private lazy var searchField = app.searchFields.firstMatch
+    private lazy var resultItem = app.staticTexts.matching(identifier: "searchResult")
+    private lazy var noResultsFoundMessage = app.staticTexts["No results found"].firstMatch
+    private lazy var clearTextButton = app.buttons["Clear text"].firstMatch
+    private lazy var recentSearchItem = app.collectionViews.cells.staticTexts
+    
+    // MARK: - Actions
+    func typeIn(searchQuery: String) {
+        searchField.typeText(searchQuery)
+    }
+    
+    func tapFirstResult() {
+        resultItem.firstMatch.tap()
+    }
+    
+    func tapClearSearchText() {
+        clearTextButton.tap()
+    }
+    
+    // MARK: - Helpers
+    func getFirstResultTitle() -> String {
+        return resultItem.firstMatch.label
+    }
+    
+    func getResultsCount() -> Int {
+        return resultItem.allElementsBoundByIndex.count
+    }
+    
+    func isNoResultsFoundMessageDisplayed() -> Bool {
+        return noResultsFoundMessage.waitForExistence(timeout: TestTimeouts.generalUiElementTimeout)
+    }
+    
+    func getRecentSearches() -> [String] {
+        return recentSearchItem.allElementsBoundByIndex.map { $0.label }
```

Comment:
This query might get you in trouble in a future. What `recentSearchItem` returns, is all `staticTexts` inside any `collectionView` - you do not specify if this is a collection view that contains the results. If, in future, any other collection view appears in the hierarchy this query will also get texts from there. 

Be more precise when narrowing down such queries.

Reply:
Yeah, I've done that to get the search results as both recent searches and search results collection view are present in the hierarchy, so for that case, I've added the accessibility identifier.